src/main/java/com/google/devtools/build/lib/metrics/MetricsCollector.java

```diff
@@ -246,11 +257,32 @@ private BuildMetrics createBuildMetrics() {
     return buildMetrics.build();
   }
 
+  private ActionData buildActionData(ActionStats actionStats) {
+    NanosToMillisSinceEpochConverter nanosToMillisSinceEpochConverter =
+        BlazeClock.createNanosToMillisSinceEpochConverter();
+    ActionData.Builder builder = ActionData.newBuilder()
+        .setMnemonic(actionStats.mnemonic)
+        .setFirstStartedMs(
+            nanosToMillisSinceEpochConverter.toEpochMillis(
+                actionStats.firstStarted.longValue()))
+        .setLastEndedMs(
+            nanosToMillisSinceEpochConverter.toEpochMillis(
+                actionStats.lastEnded.longValue()))
+        .setActionsExecuted(actionStats.numActions.get());
+    long systemTime = actionStats.systemTime.get();
+    if (systemTime > 0) {
+      builder.setSystemTime(com.google.protobuf.util.Durations.fromMillis(systemTime));
```

Comment:
Please use an import for Durations, also below.

Reply:
done